extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultTransitSecretEngine.java

```diff
@@ -0,0 +1,108 @@
+     * equivalent to:
+     * {@code encrypt(keyName, clearData.getBytes("UTF-8"), Optional.empty());}
+     * <p>
+     * this method is usually used in conjunction with {@link #decrypt(String, String)}
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param clearData the string to encrypt
+     * @return cipher text
+     */
+    String encrypt(String keyName, String clearData);
+
+    /**
+     * Encrypt a regular string with a vault key configured in the transit secret engine.
+     * if the key does not exist, and the policy specifies a create capability the key will be lazily created
+     * (i.e. upsert). the key can be further customized by specifying transit encryption-key configuration
+     * properties.
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param clearData the data to encrypt
+     * @param transitContext an optional context for key derivation
+     * @return cipher text
+     */
+    String encrypt(String keyName, BinaryData clearData, TransitContext transitContext);
+
+    /**
+     * Encrypt a list of elements. This will return a list of results, each being either a ciphertext or an error.
+     * Each element can specify an optional context, allowing key derivation.
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param requests the list of elements to encrypt
+     * @return a list of result objects containing either a ciphertext or an error
+     */
+    List<String> encrypt(String keyName, List<EncryptionRequest> requests);
+
+    // decrypt
+
+    BinaryData decrypt(String keyName, String ciphertext);
+
+    BinaryData decrypt(String keyName, String ciphertext, TransitContext transitContext);
+
+    List<DecryptionResult> decrypt(String keyName, List<DecryptionRequest> requests);
+
+    // rewrap
+
+    String rewrap(String keyName, String ciphertext);
+
+    String rewrap(String keyName, String ciphertext, TransitContext transitContext);
+
+    List<EncryptionResult> rewrap(String keyName, List<RewrappingRequest> requests);
+
+    // sign
+
+    String sign(String keyName, String input);
+
+    Signature sign(String keyName, BinaryData input, TransitContext transitContext);
+
+    List<SigningResult> sign(String keyName, SigningBatchRequest batchRequest);
+
+    // verify
+
+    boolean verify(String keyName, String signature, String input);
+
+    boolean verify(String keyName, Verifiable signature, BinaryData input, TransitContext transitContext);
+
+    List<VerificationResult> verify(String keyName, List<VerificationRequest> requests);
+
+    // keys
```

Comment:
Hmm, are these new operations ? `rotate` is not a user task. and what about those generate random functions ?

Reply:
I created `rotate` when testing the key version.
the random functions map to the [generate random bytes](https://www.vaultproject.io/api/secret/transit/index.html#generate-random-bytes) service in the transit engine.
I thought we as well add them.

Reply:
Hi Vincent @vsevel, well, IMHO neither of these functions should be in the API. The key rotation is just not a user endpoint concern, it is the same with OIDC, the endpoint only deals with the keys, the rotation is a pure administrative task which has to be securely managed, it would go out of control if all the endpoints start rotating the keys in the Vault :-)
And as for those helper functions, anyone can use some Java API to generate the random secure bytes, etc, instead of making a remote call, it is not why we integrate this transit engine :-)

Thanks

Reply:
agreed